packages/zoe/src/contracts/mintPayments.js

```diff
@@ -12,54 +13,30 @@ export const makeContract = harden(zcf => {
   // Create the internal token mint for a fungible digital asset
   const { issuer, mint, amountMath } = produceIssuer('tokens');
 
+  const zoeHelpers = makeZoeHelpers(zcf);
+
   // We need to tell Zoe about this issuer and add a keyword for the
   // issuer. Let's call this the 'Token' issuer.
   return zcf.addNewIssuer(issuer, 'Token').then(() => {
     // We need to wait for the promise to resolve (meaning that Zoe
     // has done the work of adding a new issuer).
-    const offerHook = userOfferHandle => {
+    const offerHook = offerHandle => {
       // We will send everyone who makes an offer 1000 tokens
 
-      // make the description of 1000 tokens
-      const tokenAmount = amountMath.make(1000);
-
-      // actually mint the new value of 1000 tokens
-      const tokenPayment = mint.mintPayment(tokenAmount);
-
-      // Now we must escrow the tokens with Zoe.
-      let tempContractHandle;
+      const tokens1000 = amountMath.make(1000);
+      const payment = mint.mintPayment(tokens1000);
 
-      // We need to make an invite and store the offerHandle of that
-      // invite for future use.
-      const contractSelfInvite = zcf.makeInvitation(
-        offerHandle => (tempContractHandle = offerHandle),
-      );
-      // To escrow the tokens, we must get the Zoe Service facet and
-      // make an offer
-      zcf
-        .getZoeService()
-        .offer(
-          contractSelfInvite,
-          harden({ give: { Token: tokenAmount } }),
-          // escrow the actual tokens
-          harden({ Token: tokenPayment }),
-        )
+      // Let's use a helper function which will mint a payment, escrow
```

Comment:
The helper isn't minting, right?

Reply:
Good point, that was from a previous version. I'll remove